gencode/gencode.ml

```diff
@@ -0,0 +1,266 @@
+type emoji =
+  { code_point : string
+  ; emoji : string
+  ; description : string
+  ; name : string
+  ; category : string
+  ; sub_category : string
+  }
+
+let string_escape_hex =
+  let fst_code = Char.code '0' in
+  let snd_code = Char.code 'a' - 10 in
+  let nibble_to_hex_char n =
+    if n < 0 || n >= 16 then
+      invalid_arg "nibbles must be within the range 0x0 and 0xf"
+    else if n < 10 then Char.chr (fst_code + n)
+    else Char.chr (snd_code + n)
+  in
+  fun s ->
+    let buf = Buffer.create (String.length s) in
+    String.iter
+      (fun c ->
+        let c = Char.code c in
+        Buffer.add_string buf "\\x";
+        let n = (c land 0xf0) lsr 4 in
+        Buffer.add_char buf (nibble_to_hex_char n);
+        let n = c land 0x0f in
+        Buffer.add_char buf (nibble_to_hex_char n) )
+      s;
+    Buffer.contents buf
+
+(* leading ints are illegal in
+ * OCaml identifiers so we prepend
+ * them with a '_' *)
+let wrap_leading_ints s = match s.[0] with '0' .. '9' -> "_" ^ s | _ -> s
```

Comment:
I would rather match on anything that is a valid identifier, and if it's not the case, then, add a `_`